src/Analyzer.Core/Rules/BuiltInRules.json

```diff
@@ -21,6 +21,44 @@
       "equals": true
     }
   },
+  {
+    "name": "Compute_VirtualMachine_ShouldNotBePreview",
+    "description": "Ensures that all virtual machines are not using preview images",
+    "recommendation": "Use a non-preview image for virtual machines",
+    "helpUri": "https://github.com/Azure/template-analyzer/docs/built-in-bpa-rules.md/#ensures-that-all-virtual-machines-are-not-using-preview-images",
+    "evaluation": {
+      "resourceType": "Microsoft.Compute/virtualMachines",
+      "anyOf": [
```

Comment:
Why? (not familiar with looking at these, but isn't it possible that only the version or only the reference has the preview?)

Reply:
I believe the regex enforces that "-preview" *does not* appear in the string.  So the rule should say "both paths do not have -preview".

Reply:
ah, negation!

Reply:
The issue is that when storageProfile.imageReference.version is defined, then imageReference is not a string. Regex returns false since it's not a string and then allOf is false. I have been trying to find an example where imageReference matches the regex, but I haven't found anything.

Reply:
If either field can be a string sometimes, but not always, then the rule probably needs more logic.

Reply:
Removed rule for now.